lib/crypto.ml

```diff
@@ -69,76 +63,69 @@ module Ciphers = struct
 end
 
 let sequence_buf seq =
-  let open Cstruct in
-  let buf = create 8 in
-  BE.set_uint64 buf 0 seq ;
-  buf
+  let buf = Bytes.create 8 in
+  Bytes.set_int64_be buf 0 seq ;
+  Bytes.unsafe_to_string buf
```

Comment:
allocation-wise likely a bad idea

Reply:
What do you mean? We should use `Bytes.to_string`?

Reply:
no, the entire function and allocation discipline should be different. better to allocate a bytes once, and then filling it & passing offsets along. this requires, as mentioned, a pretty deep rewriting -- which brings the suggestion to first work on benchmarks (for the latest release - e2e, maybe similar to what @reynir worked on for MirageVPN -- but would be great if cstruct allocation could be part of the benchmark), to be able to compare with the changes in here and future changes.

Reply:
> better to allocate a bytes once

Not sure to understand again, the content of this buffer actually depends on the given argument (`seq`). We probably can pre-allocate a set of `sequence_buf` and use them then but not sure this is what you want. If I think in this way, it's related to the multicore usage than can be done with `ocaml-tls`. In such context, "allocate a bytes once" (globally? per-domain?) can be risky.

Reply:
oh, sorry for the misunderstanding. my intention is: we get some data to be sent out. now, instead of allocating a header for crypto, a header for tls, etc. - let's allocate once a buffer where the encrypted payload together with the protocol header has space. thus we won't need any other allocations, neither a concat or append. this is all local to a send_data operation, not global/per domain.

nothing for this pr, though.